sky/optimizer.py

```diff
@@ -276,31 +274,40 @@ def _estimate_nodes_cost_or_time(
                             location_hint = (
                                 f' Region: {specified_resources.region}.')
 
-                    # If Kubernetes was included in the search space, then
-                    # mention "kubernetes cluster" and/instead of "catalog"
-                    # in the error message.
-                    source_hint = 'catalog'
-                    enabled_clouds = global_user_state.get_enabled_clouds()
-                    if _cloud_in_list(clouds.Kubernetes(), enabled_clouds):
-                        if specified_resources.cloud is None:
-                            source_hint = 'catalog and kubernetes cluster'
-                        elif specified_resources.cloud.is_same_cloud(
-                                clouds.Kubernetes()):
-                            source_hint = 'kubernetes cluster'
+                        # If Kubernetes was included in the search space, then
+                        # mention "kubernetes cluster" and/instead of "catalog"
+                        # in the error message.
+                        enabled_clouds = global_user_state.get_enabled_clouds()
```

Comment:
`specified_resources` was possibly unbound. Pylance caught this -- great reason to use VSCode. cc @romilbhardwaj

Reply:
Ah that's a great catch - thanks for fixing!